layermetadata/layer_metadata_restorer.go

```diff
@@ -11,8 +12,19 @@ import (
 	"github.com/buildpacks/lifecycle/platform"
 )
 
-//go:generate mockgen -package testmock -destination testmock/layer_metadata_restorer.go github.com/buildpacks/lifecycle LayerMetadataRestorer
-type LayerMetadataRestorer interface {
+type Cache interface {
+	Exists() bool
+	Name() string
+	SetMetadata(metadata platform.CacheMetadata) error
+	RetrieveMetadata() (platform.CacheMetadata, error)
+	AddLayerFile(tarPath string, sha string) error
+	ReuseLayer(sha string) error
+	RetrieveLayer(sha string) (io.ReadCloser, error)
+	Commit() error
+}
+
+//go:generate mockgen -package testmock -destination testmock/layer_metadata_restorer.go github.com/buildpacks/lifecycle/layermetadata MetaRestorer
+type MetaRestorer interface {
```

Comment:
I'm not sure that the names of the package and this interface are the best. But keeping this interface name as `LayerMetadataRestorer` inside a `layermetadata` package seems weird because whoever use it, will need to use: `layermetadata.LayerMetadataRestorer`.
It's not a good idea to call the interface `Restorer` for example because `testmock.Restorer` isn't a good name either.
Name suggestions are more than welcome 😊